projects/plugins/jetpack/extensions/extended-blocks/paid-blocks/with-upgrade-banner.jsx

```diff
@@ -70,50 +71,28 @@ export default createHigherOrderComponent(
 			setAttributes( { shouldDisplayFrontendBanner: ! hasParentBanner } );
 		}, [ setAttributes, hasParentBanner ] );
 
-		useEffect( () => {
-			const block = document.querySelector( `.wp-block[data-block="${ clientId }"]` );
-			if ( ! block ) {
-				return;
-			}
-
-			let upgradeBannerContainer = document.querySelector(
-				`.wp-block[data-block="${ clientId }"] > .jetpack-block-upgrade-banner-container`
-			);
-			if ( ! upgradeBannerContainer ) {
-				upgradeBannerContainer = document.createElement( 'div' );
-				upgradeBannerContainer.classList.add( 'jetpack-block-upgrade-banner-container' );
-				block.prepend( upgradeBannerContainer );
-			}
-
-			render(
-				<UpgradePlanBanner
-					className={ `is-${ name.replace( /\//, '-' ) }-paid-block` }
-					title={ null }
-					align={ attributes?.align }
-					visible={ isBannerVisible }
-					description={ usableBlocksProps?.description }
-					requiredPlan={ requiredPlan }
-					context={ bannerContext }
-					onRedirect={ () => trackUpgradeClickEvent( trackEventData ) }
-				/>,
-				upgradeBannerContainer
-			);
-		}, [
-			attributes?.align,
-			clientId,
-			isBannerVisible,
-			name,
-			requiredPlan,
-			trackEventData,
-			usableBlocksProps?.description,
-		] );
+		const blockProps = useBlockProps();
 
 		return (
 			<PaidBlockProvider
 				onBannerVisibilityChange={ setIsVisible }
 				hasParentBanner={ isBannerVisible }
 			>
-				<BlockEdit { ...props } />
+				<div { ...blockProps }>
+					<div className="jetpack-block-upgrade-banner-container">
```

Comment:
We no longer need this container, do we?

Reply:
I left it on purpose so that it has the same structure as before.

Reply:
The container was only needed to provide a DOM element to the `render` function. Since we don't use that anymore, we can definitely remove it.